### PR TITLE
Async Client Refactor, Documentation Updates, and Provider Fixes

### PR DESCRIPTION
**Pull request by kqlio67**
## # New provider added
   - `g4f/Provider/AiMathGPT.py` - Support for text generation, model `llama-3.1-70b`
   - `g4f/Provider/Editee.py` - Support for text generation, model `claude-3.5-sonnet, gpt-4o, gemini-pro, mistral-large`
   - `g4f/Provider/RubiksAI.py` - Support for text generation, model `llama-3.1-70b, gpt-4o-mini`
   - `g4f/Provider/Ai4Chat.py` - Support for text generation, model `gpt-4`

**Notes:**
> The service `RubiksAI` supports file uploads, but this function is not integrated into the provider `g4f/Provider/RubiksAI.py`

## # Update the documentation G4F
**docs(README.md): update code examples for clarity and correctness**
   - Replace placeholder `...` with `# Add any other necessary parameters` in code examples
   - Update model names to current models (e.g., `dall-e-3` instead of 'gemini')
   - Add print statements to display outputs in code examples

---
**docs(docs/interference.md): update usage instructions for OpenAI client**
   - Ensure the OpenAI client example includes the local base URL.
   - Confirm that the API POST request example uses the correct model name (`gpt-3.5-turbo` instead of `gpt-3.5-turbo-16k`).
---
**refactor(docs/async_client.md): use async_generate for image generation**
   - Replace `client.images.generate` with `client.images.async_generate` in asyncio.gather example
---

## # Updating the G4F
**feat(g4f/api/__init__.py): update client methods to asynchronous calls (https://github.com/xtekky/gpt4free/issues/2276#issue-2588495019), (https://github.com/xtekky/gpt4free/issues/2278#issue-2588634965)**
   - Refactor API client calls to use asynchronous methods for `chat.completions` and `images/generations`. This change improves performance and responsiveness by leveraging async capabilities of the FastAPI framework.
   - Updated `self.client` to use `AsyncClient`.
   - Changed `client.chat.completions.create` to `client.chat.completions.async_create`.
   - Changed `client.images.generate` to `client.images.async_generate`.
---

## # Update providers
**feat(g4f/Provider/Blackbox.py): enhance async response handling and cleanup**
   - Added methods for generating random IDs and cleaning responses.
   - Improved error handling in create_async_generator for API calls.
   - Refactored model handling and message formatting.
   - Added support flag for GPT-4 in class properties.
---
**feat(g4f/Provider/HuggingChat.py): add new model support for Nemotron (https://github.com/xtekky/gpt4free/pull/2277#issuecomment-2416856738)**
   - Add support for the Nvidia Llama-3.1-Nemotron-70B model
   - Updated the models list to include 'nvidia/Llama-3.1-Nemotron-70B-Instruct-HF'
   - Added a new alias for the Nemotron model

### CODE REVIEW

**Comment by comeback01** [comment]
Hello,

Thank you for your help, but it still doesn't work for me.

**Comment by kqlio67** [comment]
Hello @comeback01,

I understand that the error is still persisting for you. Actually, I've already solved this problem in my pull request, but we need to wait for the project creator, @xtekky, to review and merge these changes into the main branch.

In the meantime, to temporarily resolve the issue, I recommend rolling back to the previous version by using this command:

`pip install g4f==0.3.2.9`

This should help you work with a stable version until the fix is merged.

The solution is ready, we just need to wait for the merge. Once it's done, the error should be resolved in the latest version.

**Comment by comeback01** [comment]
It's strange, I deleted the project, then I did:
```

git clone https://github.com/kqlio67/gpt4free
docker compose up --build
```

So it's definitely your version that I'm using!

**Comment by TheFirstNoob** [comment]
@kqlio67 Hi, dont forget update docs **async_client.md** in `example-usage-with-asynciogather`
Change `task2 = client.images.generate` to `task2 = client.images.async_generate`

**Comment by UX0l0l** [comment]
I would say also update the "What's New" section in the README as it's kind of confusing and outdated. Specify the fact that o1 was added to the API, etc

**Comment by comeback01** [comment]
Hello,

Since I'm stubborn, I've run several tests, everything works as long as I don't modify "g4f/Provider/Blackbox.py".

If I modify it, I get the errors below:

`gpt4free-1  | Traceback (most recent call last):
gpt4free-1  |   File "<frozen runpy>", line 189, in _run_module_as_main
gpt4free-1  |   File "<frozen runpy>", line 112, in _get_module_details
gpt4free-1  |   File "/app/g4f/__init__.py", line 6, in <module>
gpt4free-1  |     from .models import Model
gpt4free-1  |   File "/app/g4f/models.py", line 6, in <module>
gpt4free-1  |     from .Provider import IterListProvider, ProviderType
gpt4free-1  |   File "/app/g4f/Provider/__init__.py", line 25, in <module>
gpt4free-1  |     from .Blackbox         import Blackbox
gpt4free-1  | ImportError: cannot import name 'Blackbox' from 'g4f.Provider.Blackbox' (/app/g4f/Provider/Blackbox.py)
gpt4free-1  | Traceback (most recent call last):
gpt4free-1  |   File "<frozen runpy>", line 189, in _run_module_as_main
gpt4free-1  |   File "<frozen runpy>", line 112, in _get_module_details
gpt4free-1  |   File "/app/g4f/__init__.py", line 6, in <module>
gpt4free-1  |     from .models import Model
gpt4free-1  |   File "/app/g4f/models.py", line 6, in <module>
gpt4free-1  |     from .Provider import IterListProvider, ProviderType
gpt4free-1  |   File "/app/g4f/Provider/__init__.py", line 25, in <module>
gpt4free-1  |     from .Blackbox         import Blackbox
gpt4free-1  | ImportError: cannot import name 'Blackbox' from 'g4f.Provider.Blackbox' (/app/g4f/Provider/Blackbox.py)
gpt4free-1  | 2024-10-17 12:24:06,259 WARN exited: g4f-gui (exit status 1; not expected)
gpt4free-1  | 2024-10-17 12:24:06,328 INFO spawned: 'g4f-gui' with pid 97
gpt4free-1  | 2024-10-17 12:24:06,474 WARN exited: g4f-api (exit status 1; not expected)
gpt4free-1  | 2024-10-17 12:24:06,495 INFO spawned: 'g4f-api' with pid 102
gpt4free-1  | 2024-10-17 12:24:07,547 INFO success: g4f-api entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
gpt4free-1  | 2024-10-17 12:24:07,548 INFO success: g4f-gui entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
gpt4free-1  | Traceback (most recent call last):
gpt4free-1  |   File "<frozen runpy>", line 189, in _run_module_as_main
gpt4free-1  |   File "<frozen runpy>", line 112, in _get_module_details
gpt4free-1  |   File "/app/g4f/__init__.py", line 6, in <module>
gpt4free-1  |     from .models import Model
gpt4free-1  |   File "/app/g4f/models.py", line 6, in <module>
gpt4free-1  |     from .Provider import IterListProvider, ProviderType
gpt4free-1  |   File "/app/g4f/Provider/__init__.py", line 25, in <module>
gpt4free-1  |     from .Blackbox         import Blackbox
gpt4free-1  | ImportError: cannot import name 'Blackbox' from 'g4f.Provider.Blackbox' (/app/g4f/Provider/Blackbox.py)
gpt4free-1  | Traceback (most recent call last):
gpt4free-1  |   File "<frozen runpy>", line 189, in _run_module_as_main
gpt4free-1  |   File "<frozen runpy>", line 112, in _get_module_details
gpt4free-1  |   File "/app/g4f/__init__.py", line 6, in <module>
gpt4free-1  |     from .models import Model
gpt4free-1  |   File "/app/g4f/models.py", line 6, in <module>
gpt4free-1  |     from .Provider import IterListProvider, ProviderType
gpt4free-1  |   File "/app/g4f/Provider/__init__.py", line 25, in <module>
gpt4free-1  |     from .Blackbox         import Blackbox
gpt4free-1  | ImportError: cannot import name 'Blackbox' from 'g4f.Provider.Blackbox' (/app/g4f/Provider/Blackbox.py)
gpt4free-1  | 2024-10-17 12:24:07,749 WARN exited: g4f-gui (exit status 1; not expected)
gpt4free-1  | 2024-10-17 12:24:07,818 INFO spawned: 'g4f-gui' with pid 103
gpt4free-1  | 2024-10-17 12:24:07,819 WARN exited: g4f-api (exit status 1; not expected)
^CGracefully stopping... (press Ctrl+C again to force)
[+] Stopping 1/1
 ✔ Container gpt4free-gpt4free-1  Stopped                                                                                                                                                                                               6.0s
canceled
[root@srv2402ai gpt4free]#
`

File: g4f/Provider/Ai4Chat.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+class Ai4Chat(AsyncGeneratorProvider, ProviderModelMixin):
+    url = "https://www.ai4chat.co"
+    api_endpoint = "https://www.ai4chat.co/generate-response"
+    working = True
+    supports_gpt_4 = False
+    supports_stream = False
+    supports_system_message = True
+    supports_message_history = True
+    
+    default_model = 'gpt-4'
```
Comment (johnd0e) [inline g4f/Provider/Ai4Chat.py:20]: supports_gpt_4 = False
...
default_model = 'gpt-4'
?

File: g4f/Provider/Editee.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+from .helper import format_prompt
+
+
+class Editee(AsyncGeneratorProvider, ProviderModelMixin):
+    label = "Editee"
+    url = "https://editee.com"
+    api_endpoint = "https://editee.com/submit/chatgptfree"
+    working = True
+    supports_gpt_4 = True
+    supports_stream = True
```
Comment (johnd0e) [inline g4f/Provider/Editee.py:15]: Wrong

**Pull request merged**